### PR TITLE
Add samples index

### PR DESCRIPTION
**Pull request by johnpryan**
This adds the new index and sets up `package:peanut` tooling to host the web demos alongside the samples.

[demo deployment using peanut](http://jpryan.me/samples/#)

### CODE REVIEW

**Comment by clocksmith** [comment]
Nice!

File: web/samples_index/lib/src/templates.dart [hunk 1]
```diff
@@ -0,0 +1,219 @@
+import 'dart:convert';
+
+import 'data.dart';
+import 'util.dart' as util;
+
+String _e(String s) =>
+    HtmlEscape(HtmlEscapeMode.attribute).convert(s);
+String _ea(String s) => HtmlEscape(HtmlEscapeMode.element).convert(s);
```
Comment (RedBrogdon) [inline web/samples_index/lib/src/templates.dart]: What does the "a" in `_ea` signify?

File: web/samples_index/web/add_to_app.html [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      </button>
+        </div>
+        <div class="tags-container">
+          <div class="tags-label">
+            <i class="material-icons">local_offer</i>
+            <span>Tags</span>
+          </div>
+          <div class="tags">
+            <a href="./#?search=tag%3Aadvanced">advanced</a>
+        <a href="./#?search=tag%3Asample">sample</a>
```
Comment (RedBrogdon) [inline web/samples_index/web/add_to_app.html]: Indenting looks a little weird in this file.

File: web/samples_index/web/add_a_drawer_to_a_screen.html [hunk 3]
```diff
@@ -0,0 +1,64 @@
+<!DOCTYPE html>
```
Comment (RedBrogdon) [inline web/samples_index/web/add_a_drawer_to_a_screen.html]: Are these HTML files in /web the generated ones built by the samples index grinder tasks? If so, are they intended to be regenerated each time someone goes to publish the index? And if *that's* so, could they be left out of source control?

File: web/samples_index/lib/src/templates.dart (hunk 1 above)
Comment (johnpryan) [reply web/samples_index/lib/src/templates.dart]: I think these need to be renamed, originally they were _escape and _escapeAttribute I think

File: web/samples_index/web/add_a_drawer_to_a_screen.html (hunk 3 above)
Comment (johnpryan) [reply web/samples_index/web/add_a_drawer_to_a_screen.html]: Yep, it looks like things are still working without them.

File: web/samples_index/web/add_to_app.html (hunk 2 above)
Comment (johnpryan) [reply web/samples_index/web/add_to_app.html]: there's some indentation code in `templates.dart` that might need some adjustments

File: web/samples_index/lib/src/templates.dart (hunk 1 above)
Comment (johnpryan) [reply web/samples_index/lib/src/templates.dart]: fixed

File: web/samples_index/web/add_to_app.html (hunk 2 above)
Comment (johnpryan) [reply web/samples_index/web/add_to_app.html]: these generated files aren't committed to source control anymore, so I think this is a lower priority now.

**Comment by johnpryan** [comment]
@RedBrogdon This is ready for another look.

File: web/_tool/common.dart [hunk 4]
```diff
@@ -0,0 +1,35 @@
+
```
Comment (RedBrogdon) [inline web/_tool/common.dart]: nit: copyright header

File: web/samples_index/.gitignore [hunk 5]
```diff
@@ -0,0 +1,17 @@
+# firebase public directory
```
Comment (RedBrogdon) [inline web/samples_index/.gitignore]: nit: newline

File: web/samples_index/lib/browser.dart [hunk 6]
```diff
@@ -0,0 +1 @@
+export 'src/search.dart';
```
Comment (RedBrogdon) [inline web/samples_index/lib/browser.dart]: nit: copyright header for this and any other code file.

**Comment by mono0926** [comment]
Is http://jpryan.me/samples final version URL?

I found these urls at https://github.com/flutter/samples/blob/master/web/readme.md, 

- https://flutter.github.io/samples/
- flutter.github.io/samples/#?platform=web

but both redirect to this Flutter Gallery 🤔 

image


https://flutter.github.io/samples/#/

**Pull request merged**